src/vis/density/DensityVis.tsx

```diff
@@ -0,0 +1,102 @@
+import * as React from 'react';
+import { merge, uniqueId } from 'lodash';
+import { useMemo } from 'react';
+import { VisColumn, IVisConfig, IDensityConfig, EScatterSelectSettings } from '../interfaces';
+import { DensityVisSidebar } from './DensityVisSidebar';
+// eslint-disable-next-line import/no-cycle
+import { HexagonalBin } from './HexagonalBin';
+import { HexBrushOptions } from '../sidebar/HexBrushOptions';
+import { InvalidCols } from '../general';
+import { I18nextManager } from '../../i18n/I18nextManager';
+
+interface DensityVisProps {
+  config: IDensityConfig;
+  extensions?: {
+    prePlot?: React.ReactNode;
+    postPlot?: React.ReactNode;
+    preSidebar?: React.ReactNode;
+    postSidebar?: React.ReactNode;
+  };
+  columns: VisColumn[];
+  setConfig: (config: IVisConfig) => void;
+  selectionCallback?: (ids: string[]) => void;
+  selected?: { [key: string]: boolean };
+  hideSidebar?: boolean;
+}
+
+const defaultExtensions = {
+  prePlot: null,
+  postPlot: null,
+  preSidebar: null,
+  postSidebar: null,
+};
+
+export function DensityVis({ config, extensions, columns, setConfig, selectionCallback = () => null, selected = {}, hideSidebar = false }: DensityVisProps) {
+  const mergedExtensions = useMemo(() => {
+    return merge({}, defaultExtensions, extensions);
+  }, [extensions]);
+
+  const id = React.useMemo(() => uniqueId('PCPVis'), []);
+
+  return (
+    <div className="d-flex flex-row w-100 h-100" style={{ minHeight: '0px' }}>
+      <div
+        className="position-relative d-grid flex-grow-1"
+        style={{ gridTemplateColumns: 'minmax(0, 1fr) '.repeat(config.numColumnsSelected.length < 3 ? 1 : config.numColumnsSelected.length) }}
+      >
+        {config.numColumnsSelected.length < 2 ? (
+          <InvalidCols
+            headerMessage={I18nextManager.getInstance().i18n.t('tdp:core.vis.errorHeader')}
+            bodyMessage={I18nextManager.getInstance().i18n.t('tdp:core.vis.scatterError')}
+          />
+        ) : (
+          <>
+            {config.numColumnsSelected.length > 2 ? (
+              config.numColumnsSelected.map((xCol) => {
+                return config.numColumnsSelected.map((yCol) => {
+                  if (xCol.id !== yCol.id) {
+                    return (
+                      <HexagonalBin
+                        selectionCallback={selectionCallback}
+                        selected={selected}
+                        config={config}
+                        columns={[columns.find((col) => col.info.id === xCol.id), columns.find((col) => col.info.id === yCol.id)]}
+                      />
+                    );
+                  }
+
+                  return <div key={`${xCol.id}hist`}>hello world</div>;
```

Comment:
Can we replace `hello world` with something meaningful?

Reply:
Just removed it for now. This is what is rendered in the diagonal of the small multiples. It should probably be a histogram like the scatterplot has it, but I will leave that to a future ticket

Reply:
Yes, the histogram would probably the preferred visualization in this case. Could you create a new issue then we can see if and when we want to schedule it. Thanks.